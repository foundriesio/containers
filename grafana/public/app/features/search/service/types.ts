import { DataFrame, DataSourceRef } from '@grafana/data';

export interface QueryResult {
  kind: string; // panel, dashboard, folder
  name: string;
  description?: string;
  url: string; // link to value (unique)
  tags?: string[];
  location?: LocationInfo[]; // the folder name
  datasource?: DataSourceRef[];
  score?: number;
}

export interface LocationInfo {
  kind: 'folder' | 'dashboard';
  name: string;
}

export interface QueryFilters {
  kind?: string; // limit to a single type
  tags?: string[]; // match all tags
  datasource?: string; // limit to a single datasource
}

export interface QueryResponse {
  body: DataFrame;
}

export interface GrafanaSearcher {
  search: (query: string, filter?: QueryFilters) => Promise<QueryResponse>;
}
