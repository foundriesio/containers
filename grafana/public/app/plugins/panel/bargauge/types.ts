import { SelectableValue } from '@grafana/data';
import { SingleStatBaseOptions, BarGaugeDisplayMode } from '@grafana/ui';

export interface BarGaugeOptions extends SingleStatBaseOptions {
  displayMode: BarGaugeDisplayMode;
  showUnfilled: boolean;
  minVizWidth: number;
  minVizHeight: number;
}

export const displayModes: Array<SelectableValue<string>> = [
  { value: BarGaugeDisplayMode.Gradient, label: 'Gradient' },
  { value: BarGaugeDisplayMode.Lcd, label: 'Retro LCD' },
  { value: BarGaugeDisplayMode.Basic, label: 'Basic' },
];
