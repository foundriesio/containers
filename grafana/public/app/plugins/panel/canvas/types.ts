export enum LayerActionID {
  Delete = 'delete',
  Duplicate = 'duplicate',
  MoveTop = 'move-top',
  MoveBottom = 'move-bottom',
}
