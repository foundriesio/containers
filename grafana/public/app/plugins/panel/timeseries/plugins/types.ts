interface AnnotationsDataFrameViewDTO {
  id: string;
  dashboardId: number;
  time: number;
  timeEnd: number;
  text: string;
  tags: string[];
  alertId?: number;
  newState?: string;
  title?: string;
  color: string;
  login?: string;
  avatarUrl?: string;
  isRegion?: boolean;
}
