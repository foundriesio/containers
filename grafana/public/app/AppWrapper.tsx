import React, { ComponentType } from 'react';
import { Provider } from 'react-redux';
import { Router, Route, Redirect, Switch } from 'react-router-dom';

import { config, locationService, navigationLogger } from '@grafana/runtime';
import { ErrorBoundaryAlert, GlobalStyles, ModalRoot, ModalsProvider } from '@grafana/ui';
import { SearchWrapper } from 'app/features/search';
import { getAppRoutes } from 'app/routes/routes';
import { store } from 'app/store/store';

import { AngularRoot } from './angular/AngularRoot';
import { loadAndInitAngularIfEnabled } from './angular/loadAndInitAngularIfEnabled';
import { GrafanaApp } from './app';
import { AppNotificationList } from './core/components/AppNotifications/AppNotificationList';
import { NavBar } from './core/components/NavBar/NavBar';
import { NavBarNext } from './core/components/NavBar/Next/NavBarNext';
import { I18nProvider } from './core/localisation';
import { GrafanaRoute } from './core/navigation/GrafanaRoute';
import { RouteDescriptor } from './core/navigation/types';
import { contextSrv } from './core/services/context_srv';
import { ConfigContext, ThemeProvider } from './core/utils/ConfigProvider';
import { LiveConnectionWarning } from './features/live/LiveConnectionWarning';

interface AppWrapperProps {
  app: GrafanaApp;
}

interface AppWrapperState {
  ready?: boolean;
}

/** Used by enterprise */
let bodyRenderHooks: ComponentType[] = [];
let pageBanners: ComponentType[] = [];

export function addBodyRenderHook(fn: ComponentType) {
  bodyRenderHooks.push(fn);
}

export function addPageBanner(fn: ComponentType) {
  pageBanners.push(fn);
}

export class AppWrapper extends React.Component<AppWrapperProps, AppWrapperState> {
  constructor(props: AppWrapperProps) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    await loadAndInitAngularIfEnabled();
    this.setState({ ready: true });
    $('.preloader').remove();
  }

  renderRoute = (route: RouteDescriptor) => {
    const roles = route.roles ? route.roles() : [];

    return (
      <Route
        exact={route.exact === undefined ? true : route.exact}
        path={route.path}
        key={route.path}
        render={(props) => {
          navigationLogger('AppWrapper', false, 'Rendering route', route, 'with match', props.location);
          // TODO[Router]: test this logic
          if (roles?.length) {
            if (!roles.some((r: string) => contextSrv.hasRole(r))) {
              return <Redirect to="/" />;
            }
          }

          return <GrafanaRoute {...props} route={route} />;
        }}
      />
    );
  };

  renderRoutes() {
    return <Switch>{getAppRoutes().map((r) => this.renderRoute(r))}</Switch>;
  }

  render() {
    const { ready } = this.state;

    navigationLogger('AppWrapper', false, 'rendering');

    const newNavigationEnabled = Boolean(config.featureToggles.newNavigation);

    return (
      <Provider store={store}>
        <I18nProvider>
          <ErrorBoundaryAlert style="page">
            <ConfigContext.Provider value={config}>
              <ThemeProvider>
                <ModalsProvider>
                  <GlobalStyles />
                  <div className="grafana-app">
                    <Router history={locationService.getHistory()}>
                      {ready && <>{newNavigationEnabled ? <NavBarNext /> : <NavBar />}</>}
                      <main className="main-view">
                        {pageBanners.map((Banner, index) => (
                          <Banner key={index.toString()} />
                        ))}

                        <AngularRoot />
                        <AppNotificationList />
                        <SearchWrapper />
                        {ready && this.renderRoutes()}
                        {bodyRenderHooks.map((Hook, index) => (
                          <Hook key={index.toString()} />
                        ))}
                      </main>
                    </Router>
                  </div>
                  <LiveConnectionWarning />
                  <ModalRoot />
                </ModalsProvider>
              </ThemeProvider>
            </ConfigContext.Provider>
          </ErrorBoundaryAlert>
        </I18nProvider>
      </Provider>
    );
  }
}
