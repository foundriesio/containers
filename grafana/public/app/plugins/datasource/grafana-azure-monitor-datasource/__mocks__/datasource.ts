import Datasource from '../datasource';

type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>;
};

export default function createMockDatasource(overrides?: DeepPartial<Datasource>) {
  // We make this a partial so we get _some_ kind of type safety when making this, rather than
  // having it be any or casted immediately to Datasource
  const _mockDatasource: DeepPartial<Datasource> = {
    getVariables: jest.fn().mockReturnValue([]),

    azureMonitorDatasource: {
      isConfigured() {
        return true;
      },
      getSubscriptions: jest.fn().mockResolvedValueOnce([]),
      defaultSubscriptionId: 'subscriptionId',
    },

    getAzureLogAnalyticsWorkspaces: jest.fn().mockResolvedValueOnce([]),

    getResourceGroups: jest.fn().mockResolvedValueOnce([]),
    getMetricDefinitions: jest.fn().mockResolvedValueOnce([]),
    getResourceNames: jest.fn().mockResolvedValueOnce([]),
    getMetricNamespaces: jest.fn().mockResolvedValueOnce([]),
    getMetricNames: jest.fn().mockResolvedValueOnce([]),
    getMetricMetadata: jest.fn().mockResolvedValueOnce({
      primaryAggType: 'Average',
      supportedAggTypes: ['Average', 'Maximum', 'Minimum'],
      supportedTimeGrains: [],
      dimensions: [],
    }),

    azureLogAnalyticsDatasource: {
      getKustoSchema: () => Promise.resolve(),
      getDeprecatedDefaultWorkSpace: () => 'defaultWorkspaceId',
    },
    resourcePickerData: {
      getSubscriptions: () => jest.fn().mockResolvedValue([]),
      getResourceGroupsBySubscriptionId: jest.fn().mockResolvedValue([]),
      getResourcesForResourceGroup: jest.fn().mockResolvedValue([]),
      getResourceURIFromWorkspace: jest.fn().mockReturnValue(''),
    },
    ...overrides,
  };

  const mockDatasource = _mockDatasource as Datasource;

  return jest.mocked(mockDatasource, true);
}
