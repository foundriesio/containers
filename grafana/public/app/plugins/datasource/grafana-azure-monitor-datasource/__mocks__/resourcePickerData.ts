import ResourcePicker from '../resourcePicker/resourcePickerData';

type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>;
};

export default function createMockResourcePickerData(overrides?: DeepPartial<ResourcePicker>) {
  const _mockResourcePicker: DeepPartial<ResourcePicker> = {
    getSubscriptions: () => jest.fn().mockResolvedValue([]),
    getResourceGroupsBySubscriptionId: jest.fn().mockResolvedValue([]),
    getResourcesForResourceGroup: jest.fn().mockResolvedValue([]),
    getResourceURIFromWorkspace: jest.fn().mockReturnValue(''),
    ...overrides,
  };

  const mockDatasource = _mockResourcePicker as ResourcePicker;

  return jest.mocked(mockDatasource, true);
}
