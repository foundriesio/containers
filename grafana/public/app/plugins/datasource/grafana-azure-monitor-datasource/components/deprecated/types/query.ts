import { DataQuery } from '@grafana/data';

import {
  AzureLogsQuery,
  AzureMetricQuery,
  AzureQueryType,
  AzureResourceGraphQuery,
  DeprecatedAzureQueryType,
} from '../../../types';
import { GrafanaTemplateVariableQuery } from '../../../types/templateVariables';

export interface DeprecatedAzureMonitorQuery extends DataQuery {
  queryType?: AzureQueryType | DeprecatedAzureQueryType;

  subscription?: string;

  /** ARG uses multiple subscriptions */
  subscriptions?: string[];

  azureMonitor?: AzureMetricQuery;
  azureLogAnalytics?: AzureLogsQuery;
  azureResourceGraph?: AzureResourceGraphQuery;
  grafanaTemplateVariableFn?: GrafanaTemplateVariableQuery;

  /** @deprecated App Insights/Insights Analytics deprecated in v8 */
  appInsights?: ApplicationInsightsQuery;

  /** @deprecated App Insights/Insights Analytics deprecated in v8 */
  insightsAnalytics?: InsightsAnalyticsQuery;
}

/**
 * Azure Monitor App Insights sub-query properties
 * @deprecated App Insights deprecated in v8 in favor of Metrics queries
 */
export interface ApplicationInsightsQuery {
  metricName?: string;
  timeGrain?: string;
  timeGrainCount?: string;
  timeGrainType?: string;
  timeGrainUnit?: string;
  aggregation?: string;
  dimension?: string[]; // Was string before 7.1
  dimensionFilter?: string;
  alias?: string;

  /** @deprecated Migrated to Insights Analytics query  */
  rawQuery?: string;
}

/**
 * Azure Monitor Insights Analytics sub-query properties
 * @deprecated Insights Analytics deprecated in v8 in favor of Logs queries
 */
export interface InsightsAnalyticsQuery {
  query?: string;
  resultFormat?: string;

  /** @deprecated Migrate field to query  */
  rawQueryString?: string;
}
