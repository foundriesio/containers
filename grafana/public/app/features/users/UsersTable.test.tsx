import { shallow } from 'enzyme';
import React from 'react';

import { ConfirmModal } from '@grafana/ui';
import { OrgUser } from 'app/types';

import UsersTable, { Props } from './UsersTable';
import { getMockUsers } from './__mocks__/userMocks';

jest.mock('app/core/core', () => ({
  contextSrv: {
    hasPermission: () => true,
    hasPermissionInMetadata: () => true,
    licensedAccessControlEnabled: () => false,
  },
}));

const setup = (propOverrides?: object) => {
  const props: Props = {
    users: [] as OrgUser[],
    onRoleChange: jest.fn(),
    onRemoveUser: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<UsersTable {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render users table', () => {
    const wrapper = setup({
      users: getMockUsers(5),
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Remove modal', () => {
  it('should render correct amount', () => {
    const wrapper = setup({
      users: getMockUsers(3),
    });
    expect(wrapper.find(ConfirmModal).length).toEqual(0);
  });
});
