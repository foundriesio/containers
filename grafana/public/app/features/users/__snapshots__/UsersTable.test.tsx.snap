// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<Fragment>
  <table
    className="filter-table form-inline"
  >
    <thead>
      <tr>
        <th />
        <th>
          Login
        </th>
        <th>
          Email
        </th>
        <th>
          Name
        </th>
        <th>
          Seen
        </th>
        <th>
          Role
        </th>
        <th
          style={
            Object {
              "width": "34px",
            }
          }
        />
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;

exports[`Render should render users table 1`] = `
<Fragment>
  <table
    className="filter-table form-inline"
  >
    <thead>
      <tr>
        <th />
        <th>
          Login
        </th>
        <th>
          Email
        </th>
        <th>
          Name
        </th>
        <th>
          Seen
        </th>
        <th>
          Role
        </th>
        <th
          style={
            Object {
              "width": "34px",
            }
          }
        />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0-0"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-0"
          >
            user-0
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-0@test.com"
          >
            user-0@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-0 test"
          >
            user-0 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
      <tr
        key="1-1"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-1"
          >
            user-1
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-1@test.com"
          >
            user-1@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-1 test"
          >
            user-1 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
      <tr
        key="2-2"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-2"
          >
            user-2
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-2@test.com"
          >
            user-2@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-2 test"
          >
            user-2 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
      <tr
        key="3-3"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-3"
          >
            user-3
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-3@test.com"
          >
            user-3@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-3 test"
          >
            user-3 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
      <tr
        key="4-4"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-4"
          >
            user-4
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-4@test.com"
          >
            user-4@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-4 test"
          >
            user-4 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
      <tr
        key="5-5"
      >
        <td
          className="width-2 text-center"
        >
          <img
            alt="User avatar"
            className="filter-table__avatar"
            src="url/to/avatar"
          />
        </td>
        <td
          className="max-width-6"
        >
          <span
            className="ellipsis"
            title="user-5"
          >
            user-5
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-5@test.com"
          >
            user-5@test.com
          </span>
        </td>
        <td
          className="max-width-5"
        >
          <span
            className="ellipsis"
            title="user-5 test"
          >
            user-5 test
          </span>
        </td>
        <td
          className="width-1"
        />
        <td
          className="width-8"
        >
          <OrgRolePicker
            aria-label="Role"
            disabled={false}
            onChange={[Function]}
            value="Admin"
          />
        </td>
        <td>
          <Button
            aria-label="Delete user"
            icon="times"
            onClick={[Function]}
            size="sm"
            variant="destructive"
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
